src/WidgetBase.ts

```diff
@@ -421,8 +464,8 @@ export class WidgetBase<P extends WidgetProperties = WidgetProperties, C extends
 		return this._cachedVNode;
 	}
 
-	public invalidate(): void {
-		if (this._renderState === WidgetRenderState.IDLE) {
+	public invalidate(force?: boolean): void {
```

Comment:
I don't think this is actually working as expected. Since meta relies on having access to DOM, it means it’s hitting the next `requestAnimationFrame` where the projector actually creates the DOM, which means this forced invalidate isn’t happening synchronously. If we have to render in a new animation frame anyway, we should just invalidate using that API.